backend/src/services/comparisons/comparisons.service.ts

```diff
@@ -298,6 +314,139 @@ const getComparisonOptions = async (optionType: Type): Promise<string[]> => {
+    where: { comparison_id: activeComparison.id },
+    select: {
+      complectation_id: true,
+      position: true,
+      complectation: {
+        select: {
+          name: true,
+          model_id: true,
+        },
+      },
+    },
+  });
+
+  const priceRanges = await prisma.prices_Range.findMany({
+    where: {
+      complectation_id: {
+        in: Array.from(complectations, (c) => c.complectation_id),
+      },
+    },
+    select: {
+      complectation_id: true,
+      price_start: true,
+      price_end: true,
+    },
+  });
+
+  const modelsInfo = await prisma.model.findMany({
+    where: {
+      id: { in: Array.from(complectations, (c) => c.complectation.model_id) },
+    },
+    select: {
+      id: true,
+      name: true,
+      photo_urls: true,
+      brand: {
+        select: {
+          name: true,
+        },
+      },
+    },
+  });
+
+  const complectationsInfo = await prisma.complectation.findMany({
+    where: {
+      id: { in: Array.from(complectations, (c) => c.complectation_id) },
+    },
+    select: {
+      id: true,
+      name: true,
+      users_wishlists: {
+        where: {
+          user_id: userId,
+        },
+        select: {
+          id: true,
+        },
+      },
+    },
+  });
+
+  const comparisonInfo = complectations.map((compl) => {
+    const modelIndex = modelsInfo.findIndex(
+      (model) => model.id === compl.complectation.model_id,
+    );
+    const complIndex = complectationsInfo.findIndex(
+      (complInfo) => complInfo.id === compl.complectation_id,
+    );
+    const priceIndex = priceRanges.findIndex(
+      (prices) => prices.complectation_id === compl.complectation_id,
+    );
+    return {
+      id: compl.complectation_id,
+      complectationName: compl.complectation.name,
+      position: compl.position,
+      brandName: modelsInfo[modelIndex].brand.name,
+      modelName: modelsInfo[modelIndex].name,
+      modelId: modelsInfo[modelIndex].id,
+      photos: modelsInfo[modelIndex].photo_urls,
+      priceStart: priceRanges[priceIndex].price_start,
+      priceEnd: priceRanges[priceIndex].price_end,
+      wishlistId: complectationsInfo[complIndex].users_wishlists[0]?.id,
+    };
+  });
+
+  return comparisonInfo;
+};
+
+const updatePositions = async (
+  userId: string,
+  positions: string[],
+): Promise<void> => {
+  const activeComparison = await prisma.comparison.findFirst({
+    where: {
+      active: true,
+      user_id: userId,
+    },
+    select: {
+      id: true,
+    },
+  });
+  if (!activeComparison) {
+    throw new Error('There is no active comparison');
+  }
+  positions.map(async (complectationId, index) => {
+    await prisma.comparisons_Complectations.update({
+      where: {
+        comparison_id_complectation_id: {
+          comparison_id: activeComparison.id,
+          complectation_id: complectationId,
+        },
+      },
+      data: { position: index + 1 },
+    });
+  });
```

Comment:
This can be optimized with parallel updates:
Use `await Promise.all((complectationId, index) => prisma.comparisons_Complectation.update....)`

In that way inside the `Promise.all` you'll have an array of promises that will be resolved in parallel